ircd/s_user.c

```diff
@@ -813,6 +818,31 @@ int whisper(struct Client* source, const char* nick, const char* channel,
   return 0;
 }
 
+/** Return true if the source client isn't authenticated to a registered
+ *  username, and the dest client has mode +R (block unauthed users).
+ * @param[in] source Client sending the message.
+ * @param[in] dest Destination of the message.
+ */
+int should_block_unauth_user(struct Client *source, struct Client *dest)
+{
+    // Only block regular users.
+    if (IsServer(source) || IsChannelService(source))
+        return 0;
+
+    return IsBlockUnauthUsers(dest) && !IsAccount(source);
```

Comment:
It's probably a little clearer and faster to write this whole function as (with line wrapping where appropriate):
```
return IsBlockUnauthUsers(dest) && !(IsAccount(source) || IsChannelService(source) || IsServer(source));
```